### ISSUE TITLE
Feature Request: Add wine inventory

### ISSUE THREAD

**Issue by augustf** [karlomikus/bar-assistant#119]
I wasn't going to suggest it, as the whole wine inventory function can get wildly complicated, but in light of #78, I think it could be a nice idea to add a "Wine Rack" area that allows a basic inventory function for wine bottles.

**Comment by Dogeek** [comment]
Maybe this could be more generic ? Adding a liquor subcategory so that we can track different whiskies, rhums, wines, gins etc)

**Comment by zhdenny** [comment]
Bar Assistant already provides you the tools to create your own inventory and categories however you want. And there is also already support for ingredients which are child ingredients under a bigger parent ingredient.

I already have many liquor subcategories I created myself and parent/child ingredients.

The default setup is just pretty bare bones. The tools are already in place for you to modify the whole thing.

Screenshot below is just a small section of my ingredient category page in Settings page.

![Screenshot_20230625-065511](https://github.com/karlomikus/bar-assistant/assets/26662131/6cd46461-6c48-4081-ad29-8003214c988b)

**Comment by karlomikus** [comment]
I'm adding coffee mostly because I like specialty coffee, and it's something I also like to "track" in some way.

I'd like to add specific wine and beer sections somewhere in the future, but I don't have concrete plan for that yet.

**Comment by augustf** [comment]
Right-I think the trick is adding stuff that will be separate from the ingredient/cocktails area, since they (mostly) don't interact in that way. Once there's a space to add non-cocktail things (coffee, wine, beer), it should be easy to replicate.

**Comment by malelan** [comment]
Appreciate the commentary above on how to separate ingredients into different categories, but this really only works when tying into cocktail recipes if you’re able to set an ingredient category as an ingredient for the recipe.

I.E the category could be Bourbon,  individual ingredients under that category, could be different types/brands of bourbon, however you would want to set your cocktail recipe to include 30ml of Bourbon, not a specific type/brand.  The ability to set an ingredient category isn’t available today, so storing your ingredients in the sub-category structure outlined above doesn’t really make sense, unless you are going to require specific types/brands for each individual cocktail recipe.

**Comment by malelan** [comment]
Just replying to myself here,  The key here I think is the ‘parent ingredient’ section.  Right now if you add an ingredient to your shelf, and it has a ‘parent ingredient’ if there is a cocktail that uses that ‘parent ingredient’ it will say you don’t have any, even if you have the ‘child ingredient’ on your shelf,  if this hierarchy persists, and it recognized the child ingredients as parent ingredients, it fixes the issues i mentioned above.

**Comment by karlomikus** [comment]
Tracking parent ingredient as a possible substitute was possible via ENV var, new versions now include "Track parent ingredient as a substitute ingredient in your shelf" checkbox in your profile that can toggle that behavior.

Anyhow, this is offtopic for this issue.
